fwprovider/acme/resource_acme_account.go

```diff
@@ -0,0 +1,258 @@
+func (r *acmeAccountResource) Schema(
+	_ context.Context,
+	_ resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Manages an ACME account in a Proxmox VE cluster.",
+		Attributes: map[string]schema.Attribute{
+			"contact": schema.StringAttribute{
+				Required: true,
+			},
+			"name": schema.StringAttribute{
+				Optional: true,
+			},
+			"directory": schema.StringAttribute{
+				Optional: true,
+			},
+			"eab_hmac_key": schema.StringAttribute{
+				Optional: true,
+			},
+			"eab_kid": schema.StringAttribute{
+				Optional: true,
+			},
+			"tos": schema.StringAttribute{
+				Optional: true,
+			},
+		},
+	}
+}
+
+// Configure accesses the provider-configured Proxmox API client on behalf of the resource.
+func (r *acmeAccountResource) Configure(
+	_ context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(proxmox.Client)
+	if ok {
+		r.client = *client.Cluster().ACME().Account()
+	} else {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *proxmox.Client, got: %T",
+				req.ProviderData),
+		)
+	}
+}
+
+// Create creates a new ACME account on the Proxmox cluster.
+func (r *acmeAccountResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan acmeAccountModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createRequest := &account.ACMEAccountCreateRequestBody{}
+	createRequest.Contact = plan.Contact.ValueString()
+	createRequest.Directory = plan.Directory.ValueString()
+	createRequest.EABHMACKey = plan.EABHMACKey.ValueString()
+	createRequest.EABKID = plan.EABKID.ValueString()
+	createRequest.Name = plan.Name.ValueString()
+	createRequest.TOS = plan.TOS.ValueString()
+
+	err := r.client.Create(ctx, createRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to create ACME account '%s'", plan.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read retrieves the current state of the ACME account from the Proxmox cluster.
+func (r *acmeAccountResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state acmeAccountModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	account, err := r.client.Get(ctx, state.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to read ACME account '%s'", state.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	state.Directory = types.StringValue(account.Directory)
+	state.TOS = types.StringValue(account.TOS)
+	// XXX account.Location?
+	// XXX account.Account?
```

Comment:
See account above about letsencrypt and the `data` value. `GET`ting a specific ACME account also returns the `account` object and a `location` url, which in the case of letsencrypt is a url to the letsencrypt account.
https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/acme/account/{name}

I guess these should go into the state as `computed` properties?

Reply:
In contrast to the data source (https://github.com/bpg/terraform-provider-proxmox/pull/1455#discussion_r1701744820) I think for the resource I need to extract the first email address from `data.contact` and put that into `state.Contact`, right?
Because in the `POST` https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/acme/account I'm providing `contact`, `directory`, `eab-hmac-key`, `eab-kid`, `name` and `tos_url` 
but `GET`ting an account (https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/acme/account/{name}) returns `account`, `directory`, `location` and `tos` so that should be mapped to the state, right?

Reply:
I updated the PR to take the first email address from `contact` and write that into the state.